### PR TITLE
Add Kenshi TVL

### PR DESCRIPTION
**Pull request by pouya-eghbali**
**NOTE**

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Please enable "Allow edits by maintainers" while putting up the PR.
3. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
4. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
5. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
6. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
7. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):
Kenshi

##### Twitter Link:
https://twitter.com/kenshitech

##### List of audit links if any:
https://hacken.io/audits/kenshi/

##### Website Link:
https://kenshi.io/

##### Logo (High resolution, will be shown with rounded borders):
https://kenshi.io/images/logo.svg
https://kenshi.io/images/logo/4096x4096.png

##### Current TVL:
9.16 M as reported by `node test.js projects/kenshi/index.js`

##### Treasury Addresses (if the protocol has treasury)
Protocol & Incentives: `0x4FDbc784f09BEAcEFCBDcBD7c724015416edcE53`
Treasury: `0xD59321c8266534dac369F0eFABDD5b815F1a5eb6`
Reserve: `0x51DD193630806aDCFFa9E72569a71A9c12591C33`

##### Chain:
Arbitrum

##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)
ID: `kenshi-2`
Page: https://www.coingecko.com/en/coins/kenshi

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
UCID: `27006`
Page: https://coinmarketcap.com/currencies/kenshi-v3/

##### Short Description (to be shown on DefiLlama):
Kenshi is a platform for developers that offers data indexing, oracles, blockchain IoT SDKs, crypto hardware, and a decentralized data processing network.

##### Token address and ticker if any:
Address: `0xf1264873436A0771E440E2b28072FAfcC5EEBd01`
Ticker: KNS

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Services

##### Oracle used (Chainlink/Band/API3/TWAP or any other that you are using):
Own Oracles

##### forkedFrom (Does your project originate from another project):
N/A

##### methodology (what is being counted as tvl, how is tvl being calculated):
We are counting the following:

- KNS tokens in UniSwap v3 pool
- KNS tokens staked by our users
- Treasury and Reserve: Tokens earned back by selling services to our customers
- Protocol: Tokens set aside for grants, community incentives, partners

The last two items on the list are effectively locked; as in not touched in the past 2 years since we started our project.

##### Github org/user (Optional, if your code is open source, we can track activity):
https://github.com/kenshitech/

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/kenshi exports TVL: 
        
 
 
    arbitrum                  9.20 M
    
    total                    9.20 M

File: projects/kenshi/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const { sumTokensExport } = require("../helper/unwrapLPs");
+
+const contracts = [
+  "0x4FDbc784f09BEAcEFCBDcBD7c724015416edcE53", // Protocol & Incentives
+  "0xD59321c8266534dac369F0eFABDD5b815F1a5eb6", // Treasury
+  "0xE894BD5Ec531EC8AAe856AFC3E0Fc948Ab22Efb4", // Staking
+  "0x51DD193630806aDCFFa9E72569a71A9c12591C33", // Reserve
+  "0x68C685Fd52A56f04665b491D491355a624540e85", // UniSwap v3
```
Comment (g1nt0ki) [inline projects/kenshi/index.js:8]: we cant count all liquidity on uniswap as your tvl: https://docs.llama.fi/list-your-project/what-to-include-as-tvl

File: projects/kenshi/index.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const { sumTokensExport } = require("../helper/unwrapLPs");
+
+const contracts = [
+  "0x4FDbc784f09BEAcEFCBDcBD7c724015416edcE53", // Protocol & Incentives
+  "0xD59321c8266534dac369F0eFABDD5b815F1a5eb6", // Treasury
+  "0xE894BD5Ec531EC8AAe856AFC3E0Fc948Ab22Efb4", // Staking
```
Comment (g1nt0ki) [inline projects/kenshi/index.js:6]: we can count the kenshi tokens here under staking, value in rest of the addresses can be counted towards treasury metric.

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
.

**Comment by g1nt0ki** [comment]
hi, can I close this PR

**Pull request closed without merging**